tests/test_utils.py

```diff
@@ -316,6 +340,159 @@ def check_multistream_classic(fnames):
+    ----------
+    os
+    '''
+
+    dict_global_param = {}
+    for driver in dict_s.keys():
+        # Set up subdirectories for results and states
+        result_dir = os.path.join(result_basedir, driver)
+        state_dir = os.path.join(state_basedir, driver)
+        os.makedirs(result_dir, exist_ok=True)
+        os.makedirs(state_dir, exist_ok=True)
+
+        # Fill in global parameter options
+        s = dict_s[driver]
+        dict_global_param[driver] = s.safe_substitute(
+                test_data_dir=test_data_dir,
+                result_dir=result_dir,
+                state_dir=state_dir)
+
+    return(dict_global_param)
+
+
+def parse_classic_driver_outfile_name(fname):
+    '''helper function to parse VIC classic driver output file name'''
+    resultdir, filename = os.path.split(fname)
+    prefix, suffix = os.path.splitext(filename)
+    pieces = prefix.split('_')
+    lat, lon = map(float, pieces[-2:])
+    return VICOutFile(resultdir, prefix, lat, lon, suffix)
+
+
+def check_drivers_match_fluxes(list_drivers, result_basedir):
+    ''' Check whether the flux results are similar cross multiple drivers
+
+    Parameters
+    ----------
+    list_drivers: <dict>
+        A list of driver names to be compared
+        e.g., ['classic'; 'image']
+        NOTE: must have classic driver; classic driver will be the base for
+        comparison
+    result_basedir: <str>
+        Base directory of output fluxes results; results for drivers are
+        subdirectories under the base directory
+
+    Require
+    ----------
+    glob
+    xarray
+    numpy
+    collections.namedtuple
+    parse_classic_driver_outfile_name
+    VICOutFile
+    read_vic_ascii
+    '''
+
+    # Identify all classic driver output flux files
+    try:
+        list_fnames_classic = glob.glob(os.path.join(
+                                    result_basedir, 'classic', '*'))
+    except:
+        print('Error: must have classic driver for driver-match test!')
+        raise
+
+    # Loop over all other drivers and compare with classic driver
+    for driver in list_drivers:
+        # skip classic driver
+        if driver == 'classic':
+            continue
+        # if image driver
+        if driver == 'image':
+            # load flux file
+            if len(glob.glob(os.path.join(
+                    result_basedir, driver, '*.nc'))) > 1:
+                print('Warning: more than one netCDF file found under'
+                      'directory {}'.format(result_dir))
+            fname = glob.glob(os.path.join(result_basedir, driver, '*.nc'))[0]
+            ds_image = xr.open_dataset(fname)
+           
+            import matplotlib.pyplot as plt
+            # loop over each grid cell from classic driver
+            for fname in list_fnames_classic:
+                gcell = parse_classic_driver_outfile_name(fname)
+                df_classic = read_vic_ascii(fname)
+                ds_image_cell = ds_image.sel(lat=gcell.lat, lon=gcell.lon)
```

Comment:
I would use the `method='nearest'` option here for `sel` to avoid any precision issues.